pymodbus/client/base.py

```diff
@@ -149,7 +149,7 @@ def register(self, custom_response_class: ModbusResponse) -> None:
         """
         self.framer.decoder.register(custom_response_class)
 
-    def connect(self) -> None:
+    def connect(self) -> bool:
```

Comment:
This is not correct, at the moment it depends on the implementation, some return e.g. transport, protocol, sometimes bool and sometimes none.

The correct one is None, because the user should not check it.

This btw is one of the reasons why you should not bother checking base.py, it is full of such errors.

Reply:
There are no correct answer at the moment, base.py is pure spaghetti and a mix of old/new methods which on top sometimes are async and sometimes not.

Trying to add types in this setup is a sure way to disaster ! Saying connect is a bool, is just as false as None, just leave it untyped, and exclude the whole file.

Reply:
and None is correct !! but of course the code needs to be adapted for that, which is exactly what I am doing.